grafana/resource_data_source.go

```diff
@@ -169,6 +169,7 @@ func ResourceDataSource() *schema.Resource {
 			},
 			"secure_json_data": {
 				Type:      schema.TypeList,
+				MaxItems:  1,
```

Comment:
from the `makeSecureJSONData` method, it only looks at the first element so we could reinforce the number of elements here

Reply:
I worry that this might cause confusion for others contributing to this resource.

If you feel strongly about this change, could we at least add this to `json_data` as well and add comments clarifying what the effect of this actually is?

Reply:
oh no worries, I can revert this back. when adding the new field `private_key`, I noticed we're only getting the first element of the `secure_json_data` set so if a user were to put more then one set in their config,  then only the first will get interpreted. from a user perspective, I thought it may cause confusion to be able to provide more than 1 set but not sure if that's something to really guard against.